### PR TITLE
Integrate Metrics Explorer to Dashboard

### PR DESCRIPTION
**Pull request by sonamgupta21**
# Description
Added metrics query builder to dashboards.
Default dashboards are working; previously created metrics dashboards will not work.
Raw query edit and retrieving is not working.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: static/js/common.js [hunk 1]
```diff
@@ -540,32 +540,82 @@ function runPanelAggsQuery(data, panelId, chartType, dataType, panelIndex, query
...
+    }
+    let chartType = currentPanel.chartType;
+    if (chartType === 'number'){
+        let bigNumVal = null;
+        let dataType = currentPanel.dataType;
+        let rawTimeSeriesData;
+        for (const queryData of data.queriesData) {
+            rawTimeSeriesData = await fetchTimeSeriesData(queryData);
+            const parsedQueryObject = parsePromQL(queryData.queries[0].query);
+            await addQueryElementOnAlertEdit(queryData.queries[0].name, parsedQueryObject);
```
Comment (sunitakawane) [inline static/js/common.js:567]: this function should be renamed , since it is used both in alert and dashboards

File: static/js/common.js [hunk 2]
```diff
@@ -540,32 +540,82 @@ function runPanelAggsQuery(data, panelId, chartType, dataType, panelIndex, query
...
+                $('body').css('cursor', 'default');
+            }
+            $(`#panel${panelId} .panel-body #panel-loading`).hide();   
+        } 
+    }else {
+        chartDataCollection = {};
+        if(panelId === -1){// for panel on the editPanelScreen page
+            for (const queryData of data.queriesData) {
+                const parsedQueryObject = parsePromQL(queryData.queries[0].query);
+                await addQueryElementOnAlertEdit(queryData.queries[0].name, parsedQueryObject);
```
Comment (sunitakawane) [inline static/js/common.js:591]: same comment as above

File: static/js/dashboard.js [hunk 3]
```diff
@@ -455,6 +471,41 @@ function updateTimeRangeForPanel(panelIndex) {
 }
 
 
+function updateTimeRangeForAllPanels(filterStartDate,filterEndDate){
```
Comment (sunitakawane) [inline static/js/dashboard.js:474]: how is this different from updateTimeRangeForPanels

**Comment by sunitakawane** [comment]
Can you verify if the Default dashboards will also show dynamic x-axis

File: static/js/common.js (hunk 1 above)
Comment (sonamgupta21) [reply static/js/common.js]: Updated it to  `addQueryElementForAlertAndPanel`

File: static/js/dashboard.js (hunk 3 above)
Comment (sonamgupta21) [reply static/js/dashboard.js:461]: Removed the `updateTimeRangeForPanels` function and added a new function `setTimepickerValue`, which sets the time picker when we first load the dashboards.

**Comment by sonamgupta21** [comment]
Yes, the dynamic axis works for dashboard panels.
image

**Review by sunitakawane — APPROVED** [review]

**Pull request merged**